mobify/platform-scaffold
# [WEB 1373] Implement out of stock items on Merlins PWA

Implement out of stock items on Merlins PWA

 **JIRA**: [[WEB-1373] Implement out of stock items on Merlins PWA](https://mobify.atlassian.net/browse/WEB-1373)
 **Linked PRs**: N/A

## Changes
- SFCC: Added `availability` expansion to request url to grab inventory details
- Merlin's: Parse availability information from the desktop site
- Both: Display stepper & ATC button only if item is available in stock
- Both: Add `available` to Product types

## How to test-drive this PR
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1):
    1. View [out of stock product](https://www.merlinspotions.com/books/necronomicon.html)
    1. Compare with any product that is in stock (like [this one](https://www.merlinspotions.com/books/beginner-s-guide-to-transfiguration-2.html))
**NOTE:** Stepper & ATC button does not render when the item is out of stock.

- Preview [DemandWare](https://preview.mobify.com/?url=https%3A%2F%2Fmobify-tech-prtnr-na03-dw.demandware.net%2Fon%2Fdemandware.store%2FSites-2017refresh-Site%2Fdefault%2FHome-Show&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1):
    1. You will notice that some of the HTTP requests used to fail; in this case it will result in the product being in an `out of stock` state. For a product where the HTTP request does succeed ([click here](https://mobify-tech-prtnr-na03-dw.demandware.net/s/2017refresh/44736828.html)). See that it works as before.
